### PR TITLE
can bring your own history class

### PR DESCRIPTION
**Pull request by Ppito**
Hello @Frezyx,

I need to monitor my application and keep the log history to do some stuff on it. (Like, to bring back all the logs for our support operations).
So I've moved the logic of your history implementation into a class that I can override like I want.

By the way, I saw that you were thinking about a FileManager implementation, and I think we could use this approach. What do you thinks ?

And by extension, why not offer log history implementations directly with Isar, hive or SQFlite database?

In any case, by passing through a class we'll be free to make our own implementation of the history backup.

### CODE REVIEW

File: packages/talker/lib/src/history.dart [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import 'package:talker/talker.dart';
+
+class DefaultTalkerHistory implements TalkerHistory {
+  DefaultTalkerHistory(this.settings);
+
```
Comment (Frezyx) [inline packages/talker/lib/src/history.dart:5]: Can you provide documentation ?
(Public class)

File: packages/talker/lib/src/history.dart [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      if (settings.maxHistoryItems <= _history.length) {
+        _history.removeAt(0);
+      }
+      _history.add(data);
+    }
+  }
+}
+
+abstract class TalkerHistory {
+  List<TalkerDataInterface> get history;
```
Comment (Frezyx) [inline packages/talker/lib/src/history.dart:32]: Can you provide documentation ?
(Public class)

**Review by Frezyx — CHANGES_REQUESTED** [review]

**Comment by Frezyx** [comment]
Hello @Ppito ! 
Good history functional extending. 
Please fix comments and we will merge this PR

**Comment by Ppito** [comment]
Sorry, I broke my git and pulled the wrong thing ... ignore this PR, I'll make another clean one.

**Pull request closed without merging**